extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
+            final List<String> excludes,
+            final MemTreeBuilder output
+    ) throws PermissionDeniedException, LockException {
+        List<XmldbURI> subCollections;
+        try (final Collection collection = context.getBroker().openCollection(collectionPath, LockMode.READ_LOCK)) {
+            if (collection == null) {
+                reportError(output, "Collection not found: " + collectionPath);
+                return Collections.emptyList();
+            }
+
+            if (prune) {
+                pruneCollectionEntries(collection, rootTargetAbsPath, targetDirectory, excludes, output);
+            }
+
+            for (final Iterator<DocumentImpl> i = collection.iterator(context.getBroker()); i.hasNext(); ) {
+                final DocumentImpl doc = i.next();
+                final Path targetFile = targetDirectory.resolve(doc.getFileURI().toASCIIString());
+                saveFile(targetFile, doc, startDate, output);
+            }
+
+            subCollections = new ArrayList<>(collection.getChildCollectionCount(context.getBroker()));
+            for (final Iterator<XmldbURI> i = collection.collectionIterator(context.getBroker()); i.hasNext(); ) {
+                subCollections.add(i.next());
+            }
+        }
+        return subCollections;
+    }
+
+    private void pruneCollectionEntries(
+            final Collection collection,
+            final String rootTargetAbsPath,
+            final Path targetDir,
+            final List<String> excludes,
+            final MemTreeBuilder output) {
+        try (final Stream<Path> fileStream = Files.walk(targetDir, 1)) {
+            fileStream.forEach(path -> {
+                try {
+                    // guard against deletion of output folder
+                    if (rootTargetAbsPath.startsWith(path.toString())) {
+                        return;
+                    }
+
+                    if (isExcludedPath(rootTargetAbsPath, path, excludes)) {
+                        return;
+                    }
+
+                    final String fileName = path.getFileName().toString();
+                    final XmldbURI dbname = XmldbURI.xmldbUriFor(fileName);
+                    final String currentCollection = collection.getURI().getCollectionPath();
+
+                    if (collection.hasDocument(context.getBroker(), dbname)
+                            || collection.hasChildCollection(context.getBroker(), dbname)
+                            || currentCollection.endsWith("/" + fileName)) {
+                        return;
+                    }
+
+                    // handle non-empty directories
+                    if (Files.isDirectory(path)) {
+                        deleteWithExcludes(rootTargetAbsPath, path, excludes, output);
+                    } else {
+                        Files.deleteIfExists(path);
+                        // reporting
+                        output.startElement(Sync.FileDeleteElement, null);
```

Comment:
The static-final object members don't need to be be prefixed `Sync.` as they are already in scope.